flink-java/src/main/java/org/apache/flink/api/java/typeutils/TypeExtractor.java

```diff
@@ -391,6 +392,10 @@ protected TypeExtractor() {
 				curT = typeToClass(curT).getGenericSuperclass();
 			}
 			
+			if(typeToClass(curT).equals(Tuple0.class)) {
```

Comment:
Yes it can. But in this case we are not done and do not return here. The test for `if (curT instanceof Class<?>)` is done below... This `if` is an early return that is only valid if the type is `Tuple0`. Or do I miss something?